src/utils/self_test/self_test.c

```diff
@@ -434,72 +695,1112 @@ static void print_results(struct st_latency *latencies,
+						       &ivalue, ivalue);
+				if (ret < 0) {
+					ret_value = ret;
+					goto cleanup;
+				}
+
+				/* Find range for testing */
+				percent_diff = 0.;
+				if (ivalue != 0)
+					percent_diff = (value - ivalue) /
+							ivalue;
+				percent_diff *= 100.0;
+				upper = ivalue;
+				lower = ivalue;
+
+				if (status[i].flag & TST_LOW) {
+					/* avoid checkpatch warning */
+					lower = ivalue * (1.0 - scale / 100.);
+				}
+				if (status[i].flag & TST_HIGH) {
+					/* avoid checkpatch warning */
+					upper = ivalue * (1.0 + scale / 100.);
+				}
+
+				/* Test for range */
+				passed = true;
+				results = "Passed:";
+				if (Ecfg == NULL) {
+					snprintf(range, RANGE_SIZE, " ");
+					results = " ";
+				} else if ((status[i].flag &
+					    (TST_HIGH | TST_LOW)) ==
+					    (TST_HIGH | TST_LOW)) {
+					snprintf(range, RANGE_SIZE,
+						 " Range (%6d -- %6d)"
+						 " %3d%% %5.1f%%",
+						 (int)lower, (int)upper,
+						 (int)scale, percent_diff);
+					if (!((lower <= value) &&
+					      (value <= upper))) {
+						passed = false;
+						results = "Failed:";
+						/* ret_value = -1; */
+					}
+				} else if (status[i].flag & TST_HIGH) {
+					snprintf(range, RANGE_SIZE,
+						 " Range (..    -- %6d)"
+						 "  %3d%% %5.1f%%",
+						 (int)upper, (int)scale,
+						 percent_diff);
+					if (value >= upper) {
+						passed = false;
+						results = "Failed:";
+						/* ret_value = -1; */
+					}
+				} else if (status[i].flag & TST_LOW) {
+					snprintf(range, RANGE_SIZE,
+						 " Range (%6d --   "
+						 "  ..)   %3d%%"
+						 " %5.1f%%",
+						 (int)lower, (int)scale,
+						 percent_diff);
+					if (value <= lower) {
+						passed = false;
+						results = "Failed:";
+						/* ret_value = -1; */
+					}
+				}
+
+				/* print results */
+				if (firstpass)
+					printf("\n Endpoint Result (%s)\n",
+					       status[i].description);
+				firstpass = false;
+				printf("   %s : %8d  %s %s\n",
+				       key, (int)value, results, range);
+				snprintf(compar_result, sizeof(compar_result),
+					 "%s %8d %s",
+					 results, (int)value, range);
+
+				/* Log error message */
+				if (!passed) {
+					/* avoid checkpatch warning */
+					D_INFO("%s %s range check\n",
+					       key, results);
+				}
+
+				/* Place results into result section */
+				ConfigAddString(cfg_output, result_section_name,
+						key, compar_result);
+increment_code:
+				/* loop control:*/
+				j++;
+				if (j == number_keys) {
+					j = 0;
+					number_keys = ConfigGetKeys(cfg_output,
+							    section_name,
+							    key_names, max,
+							    total_number_keys);
+					total_number_keys += number_keys;
+				}
+			}
+		} /* end of valid test loop */
+	} /* end of "for" loop */
+cleanup:
+	if (cfg_expected != NULL)
+		ConfigFree(cfg_expected);
+	return ret_value;
+}
+
+/* Place output results into output file, tagging with master key */
+static int combine_results(Config *cfg_results, char *section_name)
```

Comment:
style: all functions should look like
type<newline>
functionname()
{

Reply:
Changed.